tarantool/msgpack_ext/types/datetime.py

```diff
@@ -0,0 +1,247 @@
+from copy import deepcopy
+
+import pandas
+import pytz
+
+import tarantool.msgpack_ext.types.timezones as tt_timezones
+from tarantool.error import MsgpackError, MsgpackWarning, warn
+
+# https://www.tarantool.io/en/doc/latest/dev_guide/internals/msgpack_extensions/#the-datetime-type
+#
+# The datetime MessagePack representation looks like this:
+# +---------+----------------+==========+-----------------+
+# | MP_EXT  | MP_DATETIME    | seconds  | nsec; tzoffset; |
+# | = d7/d8 | = 4            |          | tzindex;        |
+# +---------+----------------+==========+-----------------+
+# MessagePack data contains:
+#
+# * Seconds (8 bytes) as an unencoded 64-bit signed integer stored in the
+#   little-endian order.
+# * The optional fields (8 bytes), if any of them have a non-zero value.
+#   The fields include nsec (4 bytes), tzoffset (2 bytes), and
+#   tzindex (2 bytes) packed in the little-endian order.
+#
+# seconds is seconds since Epoch, where the epoch is the point where the time
+# starts, and is platform dependent. For Unix, the epoch is January 1,
+# 1970, 00:00:00 (UTC). Tarantool uses a double type, see a structure
+# definition in src/lib/core/datetime.h and reasons in
+# https://github.com/tarantool/tarantool/wiki/Datetime-internals#intervals-in-c
+#
+# nsec is nanoseconds, fractional part of seconds. Tarantool uses int32_t, see
+# a definition in src/lib/core/datetime.h.
+#
+# tzoffset is timezone offset in minutes from UTC. Tarantool uses a int16_t type,
+# see a structure definition in src/lib/core/datetime.h.
+#
+# tzindex is Olson timezone id. Tarantool uses a int16_t type, see a structure
+# definition in src/lib/core/datetime.h. If both tzoffset and tzindex are
+# specified, tzindex has the preference and the tzoffset value is ignored.
+
+SECONDS_SIZE_BYTES  = 8
+NSEC_SIZE_BYTES     = 4
+TZOFFSET_SIZE_BYTES = 2
+TZINDEX_SIZE_BYTES  = 2
+
+BYTEORDER = 'little'
+
+NSEC_IN_SEC = 1000000000
+SEC_IN_MIN = 60
+MIN_IN_DAY = 60 * 24
+
+
+def get_bytes_as_int(data, cursor, size):
+    part = data[cursor:cursor + size]
+    return int.from_bytes(part, BYTEORDER, signed=True), cursor + size
+
+def get_int_as_bytes(data, size):
+    return data.to_bytes(size, byteorder=BYTEORDER, signed=True)
+
+def compute_offset(timestamp):
+    if timestamp.tz is None:
+        return 0
+
+    utc_offset = timestamp.tz.utcoffset(timestamp)
+    # There is no precision loss since offset is in minutes
+    return utc_offset.days * MIN_IN_DAY + utc_offset.seconds // SEC_IN_MIN
+
+def pytz_to_tarantool(timestamp):
+# Tarantool 2.10 (commit 9ee45289e01232b8df1413efea11db170ae3b3b4)
+# do not support the following pytz (version 2022.2.1) timezones
```

Comment:
It may be useful to know how to make this list: it will need to be updated or tested on a different architecture.

Reply:
Added the prinout to `validate.py` script used by `gen-timezones.sh`